soc/arm/arm/mps2/soc.c

```diff
@@ -35,6 +35,7 @@ FPGAIO_INIT(2);
 
 #define NON_SECURE_FLASH_ADDRESS	(0x100000)
 #define NON_SECURE_FLASH_OFFSET	(0x10000000)
+#define BL2_HEADER_SIZE		(0x400)
```

Comment:
I think all these 3 macros need to be documented - e.g. are they specific to Tf-M? E.g. the BL2 header size being 0x400.

Reply:
I'm not intimitely familiar with this FPGA board and image, which is a bit of an oddity with some unusual design decisions, but this code reserves space for BL2 on the second (non-secure) CPU whether the bootloader is used or not.

This is used in the `ipm_mhu_dual_core` sample, not for TF-M, which currently only makes use of a single core in existing examples for Zephyr. 

I'll document them in this PR as best I understand them, though.

Reply:
```
/* The starting address for the application image on the secondary
 * (non-secure) cpu.
 */
#define NON_SECURE_FLASH_ADDRESS	(0x100000)

/* The offset for the secondary (non-secure) cpu image, which is used
 * to determine the location of the reset vector at startuo.
 */
#define NON_SECURE_FLASH_OFFSET	        (0x10000000)

/* Space reserved for TF-M's secure bootloader on the non-secure cpu.
 * This space is reserved whether BL2 is used or not.
 */
#define BL2_HEADER_SIZE                 (0x400)
```
?